src/analysisd/cleanevent.c

```diff
@@ -67,7 +67,7 @@ int OS_CleanMSG(char *msg, Eventinfo *lf)
      * repair to only one slot so we can detect the correct date format in the next step
      * ex: Mär 02 17:30:52
      */
-    if (pieces[1] == (char) 195) {
+    if (loglen >= 5 && pieces[1] == (char) 195) {
```

Comment:
Hi @cristgl,

Why `loglen` must be greater or equal than 5?
I would say that it must be 2:

Let `pieces` be `{ 195, 164, 0 }`.
_strlen(pieces) = 2_
_loglen = strlen(pieces) + 1 = 3

_loglen = 3_ guarantees that `pieces[0]`, `pieces[1]` and `pieces[2]` are defined.
```c
if (loglen >= 3 && pieces[1] == 195 && pieces[2] == 164) { ... }
```

I think it fits. Am I wrong? Please clarify your example.

Cheers!